go/arrow/cdata/cdata_exports.go

```diff
@@ -0,0 +1,390 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cdata
+
+// #include <stdlib.h>
+// #include "arrow/c/abi.h"
+// #include "arrow/c/helpers.h"
+//
+// extern void releaseExportedSchema(struct ArrowSchema* schema);
+// extern void releaseExportedArray(struct ArrowArray* array);
+//
+// void goReleaseArray(struct ArrowArray* array) {
+//	releaseExportedArray(array);
+// }
+// void goReleaseSchema(struct ArrowSchema* schema) {
+//	 releaseExportedSchema(schema);
+// }
+import "C"
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"reflect"
+	"strings"
+	"unsafe"
+
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/apache/arrow/go/arrow/ipc"
+)
+
+func encodeCMetadata(keys, values []string) []byte {
+	if len(keys) != len(values) {
+		panic("unequal metadata key/values length")
+	}
+	npairs := len(keys)
+
+	var b bytes.Buffer
+	totalSize := 4
+	for i := range keys {
+		totalSize += 8 + len(keys[i]) + len(values[i])
+	}
+	b.Grow(totalSize)
+
+	binary.Write(&b, binary.LittleEndian, int32(npairs))
```

Comment:
Hmm, this should use native endianness. See examples here:
https://github.com/apache/arrow/blob/master/cpp/src/arrow/c/bridge_test.cc#L133-L196

Reply:
That's actually a good point! I'll fix that

Reply:
fixed now